charts/temporal/tests/deployment_test.go

```diff
@@ -104,3 +104,34 @@ func TestTemplateServerDeploymentMerging(t *testing.T) {
 	require.Equal(t, "three", deployment.Spec.Template.ObjectMeta.Labels["one"])
 	require.Equal(t, "two", deployment.Spec.Template.ObjectMeta.Labels["two"])
 }
+
+func TestTemplateServerDeploymentAnnotations(t *testing.T) {
+	// t.Parallel()
+
+	helmChartPath, err := filepath.Abs("../")
+	releaseName := "temporal"
+	require.NoError(t, err)
+
+	namespaceName := "temporal-" + strings.ToLower(random.UniqueId())
+
+	var deployment appsv1.Deployment
+
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"server.frontend.deploymentAnnotations.one":  "three",
+			"server.frontend.deploymentAnnotations.four": "four",
+			"server.deploymentAnnotations.one":           "one",
+			"server.deploymentAnnotations.two":           "two",
+		},
+		KubectlOptions:    k8s.NewKubectlOptions("", "", namespaceName),
+		BuildDependencies: true,
+	}
+
+	output := helm.RenderTemplate(t, options, helmChartPath, releaseName, []string{"templates/server-deployment.yaml"})
+
+	helm.UnmarshalK8SYaml(t, output, &deployment)
+
+	require.Equal(t, "three", deployment.ObjectMeta.Annotations["one"])
+	require.Equal(t, "two", deployment.ObjectMeta.Annotations["two"])
+	require.Equal(t, "four", deployment.ObjectMeta.Annotations["four"])
+}
```

Comment:
👍

Reply:
Did add a value test on the additional annotations as well.